### PR TITLE
dm/syncer : fix unstable test in shardddl1

### PR DESCRIPTION
**Pull request by WizardXiao**
### What problem does this PR solve? 
fix https://github.com/pingcap/ticdc/issues/3398

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to update key monitor metrics in both TiCDC document and official document

### Release note 

```release-note
Please add a release note.
If you don't think this PR needs a release note then fill it with `None`.
```

### CODE REVIEW

File: dm/dm/master/shardddl/optimist.go [hunk 1]
```diff
@@ -698,7 +698,7 @@ func (o *Optimist) handleLock(info optimism.Info, tts []optimism.TargetTable, sk
 func (o *Optimist) removeLock(lock *optimism.Lock) (bool, error) {
 	failpoint.Inject("SleepWhenRemoveLock", func(val failpoint.Value) {
 		t := val.(int)
-		log.L().Info("wait new ddl info putted into etcd",
+		log.L().Info("wait new ddl info putted into etcd in optimistic",
```
Comment (Ehco1996) [inline dm/dm/master/shardddl/optimist.go:701]: why add `in optimistic` in log can fix the unsteable tese?

**Review by lance6716 — APPROVED** [review]

**Review by Ehco1996 — APPROVED** [review]
LGTM

**Pull request merged**